airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py

```diff
@@ -151,12 +152,17 @@ async def get_modified_files(git_branch: str, git_revision: str, diffed_branch:
     If the current branch is not master, it will return the list of modified files in the current branch.
     This latest case is the one we encounter when running the pipeline locally, on a local branch, or manually on GHA with a workflow dispatch event.
     """
-    if (
-        ci_context is CIContext.MASTER
-        or ci_context is CIContext.NIGHTLY_BUILDS
-        or (ci_context is CIContext.MANUAL and git_branch == "master")
-    ):
+    # Handle cases where we're interested in the diff within the one commit at the HEAD of `master`.
+    if ci_context is CIContext.MASTER or (ci_context is CIContext.MANUAL and git_branch == "master"):
         return await get_modified_files_in_commit(git_branch, git_revision, is_local)
+
+    # Handle cases where we're interested in the diff within a commit range up to a point back in time.
+    if ci_context is CIContext.NIGHTLY_BUILDS:
+        return await get_modified_files_since(git_branch, git_revision, "1 day ago", is_local)
+    elif ci_context is CIContext.WEEKLY_BUILDS:
```

Comment:
Im not a fan of each new nightly, weekly, morningly, biweekly, monthly job having this type of dependancy.

Feels opaque and long winded

Two ideas
1. Could we reduce the context back to NIGHTLY (or remove all together)
2. Add an argument called `--modified-since` where we can pass in a value to denote how far to look back from now?

Reply:
I also agree that ideally "what get tested" should not depend on the `CIContext` value but depend on the workflow inputs.

Reply:
I'll just leave it at NIGHTLY and you can do whatever you want out of band. I don't think this code path (cron + --modified) is ever used in practice anyway.